File: 0412-fizz-buzz/0412-fizz-buzz.py
class Solution(object):
    def fizzBuzz(self, n):
        """
        :type n: int
        :rtype: List[str]
        """
        answer = []
        for i in range(1,n+1):
            result = ""
            divisible = False
            if i % 3 == 0:
                result += "Fizz"
                divisible = True
            if i % 5 == 0:
                result += "Buzz"
                divisible = True

            if(divisible):
                answer.append(result)
            else:
                answer.append(str(i))

        return answer

File: 0412-fizz-buzz/0412-fizz-buzz.cpp
class Solution {
public:
    vector<string> fizzBuzz(int n) {
        vector<string> answer;
        string word = "";
        bool divisible = false;
        
        for(int i = 1; i <= n; i++){
            if(i % 3 == 0){
                divisible = true;
                word += "Fizz";
            }
            if(i % 5 == 0){
                divisible = true;
                word += "Buzz";
            }

            if(divisible){
                answer.push_back(word);
                word = "";
                divisible = false;
            }
            else{
                answer.push_back(std::to_string(i));
            }
        }
        return answer;
        
    }
};